infra/enrolment-records/repository/src/main/java/com/simprints/infra/enrolment/records/repository/local/migration/RealmToRoomMigrationScheduler.kt

```diff
@@ -0,0 +1,82 @@
+package com.simprints.infra.enrolment.records.repository.local.migration
+
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import com.simprints.infra.logging.LoggingConstants.CrashReportTag.REALM_DB_MIGRATION
+import com.simprints.infra.logging.Simber
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class RealmToRoomMigrationScheduler @Inject constructor(
+    private val realmToRoomMigrationFlagsStore: RealmToRoomMigrationFlagsStore,
+    private val workManager: WorkManager,
+) {
+    suspend fun scheduleMigrationWorkerIfNeeded() {
+        val currentStatus = realmToRoomMigrationFlagsStore.getCurrentStatus()
+        log("Current migration status for scheduling: $currentStatus")
+
+        if (!realmToRoomMigrationFlagsStore.isMigrationGloballyEnabled() ||
+            !realmToRoomMigrationFlagsStore.canRetry() ||
+            currentStatus == MigrationStatus.COMPLETED
+        ) {
+            log(" Worker not scheduled. Migration is disabled, max retries reached or already completed.")
+            workManager.cancelUniqueWork(RealmToRoomMigrationWorker.Companion.WORK_NAME)
+            return
+        }
+
+        when (currentStatus) {
+            MigrationStatus.FAILED -> {
+                log("Migration status is FAILED and can retry. Scheduling worker.")
+                enqueueMigrationWork()
+            }
+            MigrationStatus.NOT_STARTED,
+            MigrationStatus.IN_PROGRESS,
+            -> { // If app was killed during these states
+                log("Migration status is $currentStatus. Scheduling worker to start.")
+                enqueueMigrationWork()
```

Comment:
All 3 states do the same except for the logs but we've already reported the current status. So, is there really a need for this whole `when`?